libcontainer/process_linux.go

```diff
@@ -580,7 +580,18 @@ func (p *initProcess) start() (retErr error) {
 	// cgroup. We don't need to worry about not doing this and not being root
 	// because we'd be using the rootless cgroup manager in that case.
 	if err := p.manager.Apply(p.pid()); err != nil {
-		return fmt.Errorf("unable to apply cgroup configuration: %w", err)
+		if errors.Is(err, cgroups.ErrRootless) {
```

Comment:
For the purpose of warning, I think this PR LGTM.
But further more, I think we should serialize this to `state.json`, for example with a field name `noCgroup`, it is useful for doing `runc kill`.
Please see: https://github.com/opencontainers/runc/pull/4395#discussion_r1757069353

Reply:
Yes, I was thinking about it, too, but let's do improvement at a time, shall we?

Reply:
So, we are trying to cut 1.2.0 for some time now, and this PR is a result of a recent development in the area (a regression described in #4394 (and fixed by #4395). As I noted in https://github.com/opencontainers/runc/issues/4394#issuecomment-2339406290, I'm OK with the fix in #4395, but it would be nice to also introduce a warning; this is what this PR does.

I think we can introduce `noCgroup` in runc 1.3. Feel free to open an issue about it so we won't forget.